main/main.go

```diff
@@ -374,6 +375,32 @@ func initSeedGen(ctx context.Context, c client.Client, log *logr.Logger) error {
 		return err
 	}
 
+	// Restore original pull-secret after seed creation.
+	// During seed creation, the pull-secret was removed; it needs to be restored back
+	// to allow the cluster operators to fully recover in the seed cluster.
+	log.Info("Restore original pull-secret after seed creation")
+	dockerConfigJSON, err := os.ReadFile(common.PathOutsideChroot(utils.StoredPullSecret))
+	if err != nil {
+		return fmt.Errorf("failed to read .dockerconfigjson file: %w", err)
+	}
+
+	pullSecret := &corev1.Secret{
```

Comment:
Seems to be the same part in both cases just with different data, lets add it to utils? I assume it will be used in post-pivot too in this case. WDYT?

Reply:
great idea, imo